interp/interp_test.go

```
@@ -2434,6 +2445,47 @@ func TestInterpreter_Marshal(t *testing.T) {
 		require.Equal(t, types.I32(7), v)
 	})
 
+	t.Run("function receives active context", func(t *testing.T) {
+		var got context.Context
+		setup := New(program.New(nil))
+		fn, err := setup.Marshal(func(ctx context.Context) int32 {
+			got = ctx
+			return 7
+		})
+		require.NoError(t, err)
+		setup.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `setup.Close` is not checked

(errcheck)